tests/unit/amo/sagas/testUserAbuseReports.js

```diff
@@ -0,0 +1,156 @@
+      errorHandlerId: errorHandler.id,
+      message: 'Testing',
+      user: createUserAccountResponse(),
+      ...params,
+    }));
+  }
+
+  it('calls the API for abuse', async () => {
+    const user = createUserAccountResponse({ id: 50 });
+    const message = 'I would prefer the user be XUL';
+    const response = createFakeUserAbuseReport({ message, user });
+
+    mockApi
+      .expects('reportUser')
+      .once()
+      .returns(Promise.resolve(response));
+
+    _sendUserAbuseReport({ message, user });
+
+    const expectedLoadAction = loadUserAbuseReport({
+      message: response.message,
+      reporter: response.reporter,
+      user,
+    });
+
+    await sagaTester.waitFor(expectedLoadAction.type);
+    mockApi.verify();
+
+    const loadAction = sagaTester.getCalledActions()[2];
+    expect(loadAction).toEqual(expectedLoadAction);
+  });
+
+  it('handles an empty reporter', async () => {
+    const user = createUserAccountResponse({ id: 50 });
+    const message = 'I have no reporter';
+    const response = createFakeUserAbuseReport({
+      message,
+      reporter: null,
+      user,
+    });
+
+    mockApi
+      .expects('reportUser')
+      .once()
+      .returns(Promise.resolve(response));
+
+    _sendUserAbuseReport({ message, user });
+
+    const expectedLoadAction = loadUserAbuseReport({
+      message: response.message,
+      reporter: response.reporter,
+      user,
+    });
+
+    await sagaTester.waitFor(expectedLoadAction.type);
+    mockApi.verify();
+
+    const loadAction = sagaTester.getCalledActions()[2];
+    expect(loadAction).toEqual(expectedLoadAction);
+  });
+
+  it('clears the error handler', async () => {
+    _sendUserAbuseReport();
+
+    await sagaTester.waitFor(CLEAR_ERROR);
+    expect(sagaTester.getCalledActions()[1])
+      .toEqual(errorHandler.createClearingAction());
+  });
+
+  it('dispatches an error', async () => {
+    const error = new Error('some API error maybe');
+    mockApi
+      .expects('reportUser')
+      .returns(Promise.reject(error));
+
+    _sendUserAbuseReport();
+
+    const errorAction = errorHandler.createErrorAction(error);
+    await sagaTester.waitFor(errorAction.type);
+    expect(sagaTester.getCalledActions()[2]).toEqual(errorAction);
+  });
+
+  it('resets the state when an error occurs', async () => {
+    const user = createUserAccountResponse({ id: 501 });
+    const error = new Error('some API error maybe');
+    mockApi
+      .expects('reportUser')
+      .returns(Promise.reject(error));
+
+    _sendUserAbuseReport({ user });
+
+    const abortAction = abortUserAbuseReport({ user });
+    await sagaTester.waitFor(abortAction.type);
+    expect(sagaTester.getCalledActions()[3]).toEqual(abortAction);
+  });
+
+  it('throws an error if multiple reports are submitted for the same user', async () => {
+    const user = createUserAccountResponse({ id: 50 });
+
+    _sendUserAbuseReport({
+      message: 'This add-on is malwaré!',
```

Comment:
This user...

Reply:
Sorry, copypasta from the add-ons abuse reporting tests.